### PR TITLE
ISSUE-2464:Data-driven testing style improvement

### LINKED ISSUE #2464: Data-driven testing style improvement
**Summary**

There are some testing functions written in poor styles (duplicate code, hard to maintain ...), we must improve it!

Refer: https://dave.cheney.net/2019/05/07/prefer-table-driven-tests

Good styles example:

https://github.com/datafuselabs/databend/blob/main/common/functions/src/scalars/arithmetics/arithmetic_test.rs

Bad styles example:

1. https://github.com/datafuselabs/databend/blob/main/common/functions/src/scalars/dates/date_function_test.rs
2. https://github.com/datafuselabs/databend/blob/main/common/datavalues/src/arrays/arithmetic_test.rs

### PR DESCRIPTION
**Pull request by dust1**
I hereby agree to the terms of the CLA available at: https://databend.rs/policies/cla/

## Summary

Summary about this PR

## Changelog

- Improvement

## Related Issues

Fixed #2464

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

File: common/functions/src/scalars/dates/number_function.rs [hunk 1]
```diff
@@ -318,7 +318,7 @@ impl NumberResultFunction<u16> for ToMonday {
     const IS_DETERMINISTIC: bool = true;
 
     fn return_type() -> Result<DataType> {
-        Ok(DataType::Date16)
+        Ok(DataType::UInt16)
```
Comment (sundy-li) [inline common/functions/src/scalars/dates/number_function.rs:321]: This is the logical data_type, must be Date16 for toMonday.

**Comment by sundy-li** [comment]
Good job, −1,580 LOC is good win, zen programmers delete codes.

File: common/functions/src/scalars/dates/number_function.rs (hunk 1 above)
Comment (dust1) [reply common/functions/src/scalars/dates/number_function.rs:321]: Ok i should go back

**Review by sundy-li — APPROVED** [review]

**Comment by BohuTANG** [comment]
@dust1 
Thank you!
Great job :100:

**Pull request merged**